backend/de.metas.business/src/main/java/de/metas/order/model/interceptor/C_OrderLine.java

```diff
@@ -126,6 +126,16 @@ public void setQtyEnteredInPriceUOM(final I_C_OrderLine orderLine)
 		orderLine.setQtyEnteredInPriceUOM(qtyEnteredInPriceUOM);
 	}
 
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW })
+	public void setProjectToOrderline(final I_C_OrderLine orderline)
+	{
+		final I_C_Order order = orderline.getC_Order();
+		if (order != null)
+		{
+			orderline.setC_Project_ID(order.getC_Project_ID());
+		}
```

Comment:
```suggestion
		if (orderline.getC_Order_ID() > 0 && orderLine.getC_Project_ID() <= 0)
		{
			final I_C_Order order = orderline.getC_Order();
			orderline.setC_Project_ID(order.getC_Project_ID());
		}
```

improvements:
* don't overwrite a rpoject that a used just set in the add-new modal
* don't load the order if you don't have to

Reply:
applied changes.